gen/build_deploy/bash.py

```diff
@@ -361,6 +361,48 @@ def validate_custom_check_bins_dir(custom_check_bins_dir):
     fi
 }
 
+function d_type_enabled_if_xfs()
+{
+    # Return 1 if $1 is a directory on XFS volume with ftype ! = 1
+    # otherwise return 0
+    DIRNAME="$1"
+
+    RC=0
+    if [[ ! -d "$DIRNAME" ]]; then
+        return $RC
+    fi
+    read -r filesystem_device filesystem_type <<<"$(df -PT "$DIRNAME" | awk 'END{print $1,$2}')"
+    if [[ "$filesystem_type" == "xfs" ]]; then
+        echo -n -e "Checking if $DIRNAME is mounted with \"fytpe=1\": "
+        ftype_value="$(xfs_info $filesystem_device | grep -oE ftype=[0-9])"
+        if [[ "$ftype_value" != "ftype=1" ]]; then
+            RC=1
+        fi
+        print_status $RC "${NORMAL}(${ftype_value})"
+    fi
+    return $RC
+}
+
+# check node storage has d_type (ftype=1) support enabled if using XFS
+function check_xfs_ftype() {
+    RC=0
+
+    # If /var/lib/mesos exists check xfs attributes for it
+    ( d_type_enabled_if_xfs /var/lib/mesos ) || RC=1
+
+    # If docker root directory exists check xfs attributes for it
+    docker_root_dir="$(docker info -f '{{{{.DockerRootDir}}')"
+    ( d_type_enabled_if_xfs "$docker_root_dir" ) || RC=1
+
+    # If /var/lib/mesos doesn't exist then check /var/lib (parent directory) for ftype
+    if [[ ! -d /var/lib/mesos  ]]; then
+        ( d_type_enabled_if_xfs /var/lib ) || RC=1
+    fi
```

Comment:
It looks like this stanza is related to the one starting at line 390, and should be relocated next to it or merged with it. We could also DRY the `/var/lib/mesos` that gets repeated. How about something like this:

```
mesos_dir=/var/lib/mesos
if [[ -d $mesos_dir ]]; then
    ( d_type_enabled_if_xfs $mesos_dir ) || RC=1
else
    # $mesos_dir doesn't exist yet, so we check its parent dir (assumed to exist)
    ( d_type_enabled_if_xfs $(dirname $mesos_dir) ) || RC=1
fi
```